perfscale_regression_ci/scripts/network/network_policy_scalability.sh

```diff
@@ -0,0 +1,150 @@
+#/!/bin/bash
+################################################
+## Auth=mifiedle@redhat.com lhorsley@redhat.com
+## Desription: On a 10 node OVN cluster, time how long it takes to scale up to 2000 pods with and without the networkpolicy 
+## Note: The scale up time with the networkpolicy in place will be more than without, but it should not be an order of magnitude difference.	
+## This automated test does not cover step 5.1 - 5.4 in the Polarian test case
+## Polarion test case: OCP-41535 - NetworkPolicy scalability - 2000 pods per namespace using customer network policy	
+## https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-41535
+## Cluster config: Cluster needed in AWS EC2 (m5.xlarge, 3 master/etcd, 3 worker/compute nodes)
+## kube-burner config: perfscale_regerssion_ci/kubeburner-object-templates/node-affinity-anti-affinity-config.yml
+################################################ 
+
+source ../../utils/run_workload.sh
+source ../custom_workload_env.sh
+source ../common.sh
+source network_policy_scalability_env.sh
+
+pod_scale_max=2000
+pod_scale_min=500
+pass_fail=0
+project_name="np-issue-test"
+deployment_name=$project_name
+export NAME=$project_name
+export NAMESPACE=$project_name
+
+
+project_label="test=np-issue-test"
+
+
+
+echo "Create and label new project"
+prepare_project ${NAMESPACE} $project_label
+
+
+echo "Applying netowrk policy..."
+oc apply -f  ${NETWORK_POLICY} -n ${NAMESPACE}
+echo ""
+
+echo "Starting scale-up test with network policy..."
+echo ""
+
+# SCALE UP DEPLOYMENT
+scale_w_np_start_time=`date +%s`
+
+echo "======Use kube-burner to load the cluster with test objects======"
+run_workload
+
+scale_up_np=$(oc get deployment $deployment_name --no-headers -n ${NAMESPACE} | awk -F ' {1,}' '{print $4}')
+
+#scale_up_np=$(oc get pods -n ${NAMESPACE} --no-headers | wc -l)
+
+echo "start time: $scale_w_np_start_time"
+check_deployment_pod_scale ${NAMESPACE} $deployment_name $scale_up_np $pod_scale_max
```

Comment:
I think you should be able to `export WAIT_FOR=["Deployment"]` that is built into the run_workload instead of a separate script 

https://github.com/cloud-bulldozer/kube-burner/blob/master/pkg/burner/waiters.go

Reply:
because the _run_workload_ does a lot more (like indexing benchmark metadata and doing a touchstone compare) I think we should try to keep it consistent between the 2 scale steps we are comparing. So I think it would be good to create the deployment with 500 pods as part of the run workload, then start your timer and scale the deployment to 2000 like you are doing without the network policy.